android/app/src/main/java/org/fossasia/openevent/viewmodels/AboutFragmentViewModel.java

```diff
@@ -0,0 +1,89 @@
+import org.threeten.bp.format.DateTimeParseException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.RealmResults;
+
+public class AboutFragmentViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<List<Object>> sessions;
+    private MutableLiveData<ArrayList<String>> dateList;
+    private MutableLiveData<Event> event;
+    private RealmResults<Session> bookmarksResult;
+
+    public AboutFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Event> getEvent() {
+        if (event == null) {
+            event = new MutableLiveData<>();
+            event.setValue(realmRepo.getEvent());
+        }
+        return event;
+    }
+
+    public LiveData<ArrayList<String>> getDateList() {
+        if (dateList == null) {
+            dateList = new MutableLiveData<>();
+            RealmResults<EventDates> eventDates = realmRepo.getEventDatesSync();
+            ArrayList<String> dateListString = new ArrayList<>();
+            for (EventDates eventDate : eventDates) {
+                dateListString.add(eventDate.getDate());
+            }
+            dateList.setValue(dateListString);
+        }
+        return dateList;
+    }
+
+    public LiveData<List<Object>> getSessions(ArrayList<String> dateList) {
+        if (sessions == null) {
+            sessions = new MutableLiveData<>();
+            bookmarksResult = realmRepo.getBookMarkedSessions();
+            bookmarksResult.addChangeListener((bookmarked, orderedCollectionInnerChangeSet) -> {
+                List<Object> sessionsList = new ArrayList<>();
+                for (String eventDate : dateList) {
+                    boolean headerCheck = false;
+                    for (Session bookmarkedSession : bookmarked) {
+                        if (bookmarkedSession.getStartDate() != null && bookmarkedSession.getStartDate().equals(eventDate)) {
+                            if (!headerCheck) {
+                                String headerDate = "Invalid";
+                                try {
+                                    headerDate = DateConverter.formatDay(eventDate);
+                                } catch (DateTimeParseException e) {
+                                    e.printStackTrace();
+                                }
+                                sessionsList.add(headerDate);
+                                headerCheck = true;
+                            }
+                            sessionsList.add(bookmarkedSession);
+                        }
+                    }
+                }
+                sessions.setValue(sessionsList);
+            });
+        }
+        return sessions;
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        if (bookmarksResult != null) {
+            bookmarksResult.removeAllChangeListeners();
+        }
+    }
```

Comment:
All realm instances used should be stored globally and cleared here

Reply:
@iamareebjamal What are the other realm instances that should be stored globally and cleared. 
In my opinion only bookmarksResult needs to be stored as a global variable and cleared. Other realm instances can be local since they don't have any listeners attached to them. For eg in getDateList and getEvent method.

Reply:
Alright yeah :+1: